packages/nouns-contracts/contracts/StreamEscrow.sol

```diff
@@ -0,0 +1,307 @@
+    uint48 public lastForwardTimestamp;
+
+    // @dev a mapping of how much ethPerTick will end at this tick
+    mapping(uint256 tick => uint128 ethPerTick) public ethStreamEndingAtTick;
+    mapping(uint256 streamId => Stream) internal streams;
+    mapping(address => bool) public allowedToCreateStream;
+
+    /**
+     * @notice Initializes the StreamEscrow contract
+     * @param daoExecutor_ The address of the DAO executor contract
+     * @param ethRecipient_ The address that will receive ETH payments
+     * @param nounsRecipient_ The address that will receive Nouns tokens when streams are canceled
+     * @param nounsToken_ The address of the Nouns ERC721 token contract
+     * @param streamCreator_ The address that will be initially allowed to create streams
+     */
+    constructor(
+        address daoExecutor_,
+        address ethRecipient_,
+        address nounsRecipient_,
+        address nounsToken_,
+        address streamCreator_
+    ) {
+        daoExecutor = daoExecutor_;
+        ethRecipient = ethRecipient_;
+        nounsRecipient = nounsRecipient_;
+        nounsToken = INounsToken(nounsToken_);
+        allowedToCreateStream[streamCreator_] = true;
+    }
+
+    /**
+     * @notice Forwards all streams and creates a new stream for a Noun.
+     * @notice ETH value must be sent with this function call.
+     * @dev Combines forwardAll() and createStream() operations into a single transaction.
+     * @param nounId The ID of the Noun token for which the stream is being created.
+     * @param streamLengthInTicks The duration of the stream in ticks. 1 day must pass for a tick to increase.
+     */
+    function forwardAllAndCreateStream(uint256 nounId, uint16 streamLengthInTicks) external payable {
+        forwardAll();
+
+        createStream(nounId, streamLengthInTicks);
+    }
+
+    /**
+     * @notice Creates a new ETH stream for a specific Noun token.
+     * @dev Only allowed addresses that are also Noun owners/approved operators can create streams.
+     * @param nounId The ID of the Noun token to create a stream for.
+     * @param streamLengthInTicks The duration of the stream in ticks.
+     * @custom:throws 'not allowed' if sender is not allowed to create streams.
+     * @custom:throws 'only noun owner or approved' if sender is not owner/approved.
+     * @custom:throws 'stream active' if a stream already exists for the Noun.
+     * @notice ETH value must be sent with this function call.
+     */
+    function createStream(uint256 nounId, uint16 streamLengthInTicks) public payable {
+        require(allowedToCreateStream[msg.sender], 'not allowed');
+        require(isApprovedOrOwner(msg.sender, nounId), 'only noun owner or approved');
+        require(!isStreamActive(nounId), 'stream active');
+
+        // register new stream
+        uint128 ethPerTick = toUint128(msg.value / streamLengthInTicks);
+        uint32 streamLastTick = currentTick + streamLengthInTicks; // streamLastTick is inclusive
```

Comment:
I believe this line assumes `forwardAll` is called before `createStream` in `forwardAllAndCreateStream`, otherwise we would be streaming one tick too many? -- if so, I think it's very helpful to say this explicitly in the inline comment (which I'd also move to be above the code statement), for whomever might want to change the code, or it'll just help people make sense of it a bit faster.

wdyt?

Reply:
I don't think it's for that reason.
it's because forwardAll finishes streams as soon as the tick reach this streamLastTick.
I think the code has changed since I wrote this comment. at this point this comment is just confusing I think.

it's maybe a bit confusing but I'm not sure what documentation can make it less confusing. I think it's clear what the intention is, i.e. how much should be stream over what period.

Reply:
are you saying if someone changes the order of functions such that createStream is called before forwardAll, nothing will change?